### ISSUE TITLE
Integration PHPSA & ATA[0000] Error response from daemon: 400 Bad Request: malformed Host header 

### ISSUE THREAD

**Issue by ovr** [codeclimate/codeclimate#497]
Hi there!

Thanks CodeClimate Team and all Contributors on this projects

I am going to integrate https://github.com/ovr/phpsa (PHP Smart Analyzer) with CodeClimate but on my way I am having a trouble, i am not ruby developer and cannot debug code
It's an initial step for integration, PR https://github.com/ovr/phpsa/pull/210

````sh
root@dmtry /home/web/projects/phpsa # docker pull codeclimate/codeclimate
Using default tag: latest
latest: Pulling from codeclimate/codeclimate
Digest: sha256:ce2349b27f2f9e9fcb4c17f6f84bbb72a6fcf634928abe28b6db53a255ec0d94
Status: Image is up to date for codeclimate/codeclimate:latest
root@dmtry /home/web/projects/phpsa # codeclimate engines:install
Pulling docker images.
FATA[0000] Error response from daemon: 400 Bad Request: malformed Host header 
error: (CC::CLI::Engines::Install::ImagePullFailure) unable to pull image codeclimate/codeclimate-fixme
```

```sh
root@dmtry /home/web/projects/phpsa # CODECLIMATE_DEBUG=1 codeclimate analyze --dev -v
[DEBUG] Couldn't include because part of path doesn't exist. path="./-v"
Starting analysis
[DEBUG] phpsa:stable engine config: {"enabled":true,"include_paths":["./"]}
[DEBUG] docker run: ["docker", "run", "--name", "cc-engines-phpsa-stable-caa1acc0-3e37-4de8-a754-f050734c9411", "--cap-drop", "all", "--label", "com.codeclimate.label=906c35c7-ed6e-46c8-ae87-a847bf941d65", "--memory", "512000000", "--memory-swap", "-1", "--net", "none", "--rm", "--volume", "/home/web/projects/phpsa:/code:ro", "--volume", "/tmp/cc/6ba79c07-502d-417a-b26f-c8955f1303d8:/config.json:ro", "--user", "9000:9000", "codeclimate/codeclimate-phpsa:latest"]
error: (CC::Analyzer::Engine::EngineFailure) engine phpsa:stable failed with status 1 and stderr 
time="2016-10-08T05:11:57Z" level=fatal msg="Error response from daemon: 400 Bad Request: malformed Host header" 
[DEBUG] backtrace: /usr/src/app/lib/cc/analyzer/raising_container_listener.rb:23:in `finished'
	/usr/src/app/lib/cc/analyzer/composite_container_listener.rb:17:in `block in finished'
	/usr/src/app/lib/cc/analyzer/composite_container_listener.rb:17:in `each'
	/usr/src/app/lib/cc/analyzer/composite_container_listener.rb:17:in `finished'
	/usr/src/app/lib/cc/analyzer/container.rb:78:in `run'
	/usr/src/app/lib/cc/analyzer/engine.rb:52:in `run'
	/usr/src/app/lib/cc/analyzer/engines_runner.rb:60:in `block in run_engine'
	/usr/src/app/lib/cc/analyzer/formatters/plain_text_formatter.rb:74:in `with_spinner'
	/usr/src/app/lib/cc/analyzer/formatters/plain_text_formatter.rb:55:in `block in engine_running'
	/usr/src/app/lib/cc/analyzer/formatters/formatter.rb:18:in `engine_running'
	/usr/src/app/lib/cc/analyzer/formatters/plain_text_formatter.rb:54:in `engine_running'
	/usr/src/app/lib/cc/analyzer/engines_runner.rb:59:in `run_engine'
	/usr/src/app/lib/cc/analyzer/engines_runner.rb:23:in `block in run'
	/usr/src/app/lib/cc/analyzer/engines_runner.rb:23:in `each'
	/usr/src/app/lib/cc/analyzer/engines_runner.rb:23:in `run'
	/usr/src/app/lib/cc/cli/analyze.rb:20:in `block in run'
	/usr/src/app/lib/cc/cli/analyze.rb:18:in `chdir'
	/usr/src/app/lib/cc/cli/analyze.rb:18:in `run'
	/usr/src/app/lib/cc/cli/command.rb:24:in `execute'
	/usr/src/app/lib/cc/cli/runner.rb:22:in `run'
	/usr/src/app/lib/cc/cli/runner.rb:8:in `run'
	/usr/src/app/bin/codeclimate:6:in `<main>'
```

```sh
root@dmtry /home/web/projects/phpsa #   uname -a
Linux dmtry.me 4.4.0-22-generic #40-Ubuntu SMP Thu May 12 22:03:46 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
root@dmtry /home/web/projects/phpsa #   docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6.1
 Git commit:   20f81dd
 Built:        Wed, 20 Apr 2016 14:19:16 -0700
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.6.1
 Git commit:   20f81dd
 Built:        Wed, 20 Apr 2016 14:19:16 -0700
 OS/Arch:      linux/amd64
root@dmtry /home/web/projects/phpsa #   env | grep "^DOCKER_"

root@dmtry /home/web/projects/phpsa # cat /etc/issue
Ubuntu 16.04.1 LTS \n \l
```

Please help with advice for me, if any additional information is needed I can provide it

Thanks

**Comment by wfleming** [comment]
Hi @ovr,

It looks like you're running into the same issue as #475. We have a docker client embedded in the CLI, and versions of Docker prior to 1.12.0 have this unfortunate behavior when the client & daemon are compiled with different versions of Go.

This bug in Docker was fixed in 1.12.0, so the recommended remediation is to upgrade your system to Docker 1.12.0.

Hope that helps, please let us know if upgrading Docker doesn't help.

**Comment by ovr** [comment]
Thanks @wfleming, I updated docker from original PPA for ubuntu from Docker team
I am having `1.12.1, build 23cf638` now, and it's works
Thanks
